TicketBooking/Flight/FlightRepository/FlightRepository.cs

```diff
@@ -0,0 +1,174 @@
+                var values = line.Split(',');
+                var (
+                    id,
+                    departureCountry,
+                    destinationCountry,
+                    flightNo,
+                    departureDate,
+                    departureAirport,
+                    arrivalAirport) = (
+                        int.Parse(values[0]),
+                        values[1],
+                        values[2],
+                        values[6],
+                        DateTime.Parse(values[7]),
+                        values[8],
+                        values[9]);
+
+                Dictionary<string, double> flightClassesDict = new Dictionary<string, double>();
+
+                if (!string.IsNullOrWhiteSpace(values[3])) flightClassesDict.Add("FirstClass", double.Parse(values[3]));
+
+                if (!string.IsNullOrWhiteSpace(values[4])) flightClassesDict.Add("Business", double.Parse(values[4]));
+
+                if (!string.IsNullOrWhiteSpace(values[5])) flightClassesDict.Add("Economy", double.Parse(values[5]));
+
+                var validator = new FlightValidator(Flights);
+                string validationResult = validator.ValidateFlightProperties(
+                    id,
+                    departureCountry,
+                    destinationCountry,
+                    flightClassesDict,
+                    flightNo,
+                    departureDate,
+                    departureAirport,
+                    arrivalAirport);
+
+                if (validationResult.Count() == 0)
+                {
+                    Dictionary<ClassEnum, double> classesDict = new Dictionary<ClassEnum, double>();
+
+                    foreach (var c in flightClassesDict)
+                    {
+                        classesDict.Add((ClassEnum)Enum.Parse(typeof(ClassEnum), c.Key), c.Value);
+                    }
+
+                    var flight = new FlightModel.Flight()
+                    {
+                        Id = id,
+                        DepartureCountry = (CountryEnum)Enum.Parse(typeof(CountryEnum), departureCountry),
+                        DestinationCountry = (CountryEnum)Enum.Parse(typeof(CountryEnum), destinationCountry),
+                        Class = classesDict,
+                        FlightNo = flightNo,
+                        DepartureDate = departureDate,
+                        DepartureAirport = (AirportEnum)Enum.Parse(typeof(AirportEnum), departureAirport),
+                        ArrivalAirport = (AirportEnum)Enum.Parse(typeof(AirportEnum), arrivalAirport),
+                    };
+
+                    Flights.Add(flight);
+                    Console.WriteLine($"Flight holding the id: {id} uploaded successfully!");
+                }
+                else
+                {
+                    Console.WriteLine($"About flight holding the id: {id}");
+                    Console.WriteLine(validationResult);
+                }
```

Comment:
Prefer an early return and avoid using the `Count()` LINQ method whenever you have the `Count` or `Length` properties available, as the `Count()` will cause an iteration over the list, while the properties already have the value you are looking for.

```csharp
if (validationResult.Length > 0)
{
    Console.WriteLine($"About flight holding the id: {id}");
    Console.WriteLine(validationResult);
    return;
}

Dictionary<ClassEnum, double> classesDict = new();

foreach (var c in flightClassesDict)
{
    classesDict.Add((ClassEnum)Enum.Parse(typeof(ClassEnum), c.Key), c.Value);
}

var flight = new FlightModel.Flight()
{
    Id = id,
    DepartureCountry = (CountryEnum)Enum.Parse(typeof(CountryEnum), departureCountry),
    DestinationCountry = (CountryEnum)Enum.Parse(typeof(CountryEnum), destinationCountry),
    Class = classesDict,
    FlightNo = flightNo,
    DepartureDate = departureDate,
    DepartureAirport = (AirportEnum)Enum.Parse(typeof(AirportEnum), departureAirport),
    ...
}
```